hummingbot/connector/exchange/ascend_ex/ascend_ex_exchange.py

```diff
@@ -339,31 +338,36 @@ def _format_trading_rules(self, instruments_info: Dict[str, Any]) -> Dict[str, A
         return trading_rules
 
     async def _update_account_data(self):
-        headers = {
-            **self._ascend_ex_auth.get_headers(),
-            **self._ascend_ex_auth.get_auth_headers("info"),
-            **self._ascend_ex_auth.get_hb_id_headers(),
-        }
-        url = f"{CONSTANTS.REST_URL}/info"
-        response = await self._shared_client.get(url, headers=headers)
+        url = f"{CONSTANTS.REST_URL}/{CONSTANTS.INFO_PATH_URL}"
+        headers = self._ascend_ex_auth.get_headers(CONSTANTS.INFO_PATH_URL)
 
-        try:
-            parsed_response = json.loads(await response.text())
-        except Exception as e:
-            raise IOError(f"Error parsing data from {url}. Error: {str(e)}")
-        if response.status != 200:
-            raise IOError(
-                f"Error fetching data from {url}. HTTP status is {response.status}. " f"Message: {parsed_response}"
-            )
-        if parsed_response["code"] != 0:
-            raise IOError(f"{url} API call failed, response: {parsed_response}")
+        rest_assistant = await self._get_rest_assistant()
 
-        self._account_group = parsed_response["data"]["accountGroup"]
-        self._account_uid = parsed_response["data"]["userUID"]
+        request = RESTRequest(method=RESTMethod.GET,
+                              url=url,
+                              headers=headers,
+                              is_auth_required=True)
+
+        async with self._throttler.execute_task(CONSTANTS.INFO_PATH_URL):
+            response = await rest_assistant.call(request)
+
+            try:
+                parsed_response = await response.json()
+            except Exception as e:
+                raise IOError(f"Error parsing data from {url}. Error: {str(e)}")
+            if response.status != 200:
+                raise IOError(
+                    f"Error fetching data from {url}. HTTP status is {response.status}. " f"Message: {parsed_response}"
+                )
+            if parsed_response["code"] != 0:
+                raise IOError(f"{url} API call failed, response: {parsed_response}")
+
+            self._account_group = parsed_response["data"]["accountGroup"]
+            self._account_uid = parsed_response["data"]["userUID"]
```

Comment:
Unless there is a very compelling reason to implement this method like this, it should be changed make the request using `_api_request`

Reply:
_update_account_data() is being called from _api_request(), so it would result in an infinite loop